### ISSUE TITLE
Faulty implementation of pos_weight kwarg in paddle.fft.fft

### ISSUE THREAD

**Issue by KevinUli** [PaddlePaddle/Paddle#54730]
### bug描述 Describe the Bug

The implementation of pos_weight kwarg in paddle.fft.fft is incorrect. This can be seen from the code:
```
>>> import torch
>>> arr = torch.as_tensor([0.5, 0.5,], dtype=torch.float32)
>>> pos_weight = torch.as_tensor([1., 2.,], dtype=torch.float32)
>>> torch.nn.functional.binary_cross_entropy_with_logits(arr, arr, weight=arr, reduction="none", pos_weight=pos_weight)
Tensor([0.3620, 0.4806])
```

```
>>> import paddle
>>> arr = paddle.to_tensor([0.5, 0.5,], dtype="float32")
>>> pos_weight = paddle.to_tensor([1., 2.,], dtype="float32")
>>> paddle.nn.functional.binary_cross_entropy_with_logits(arr, arr, weight=arr, reduction="none", pos_weight=pos_weight)
Tensor(shape=[2], dtype=float32, place=Place(cpu), stop_gradient=True, [0.36203849, 0.54305774])
```

### 其他补充信息 Additional Supplementary Information
Torch implementation here: (https://github.com/pytorch/pytorch/blob/01abccf63faa14b05ffee2474b1fcd7d85d44782/aten/src/ATen/native/Loss.cpp#LL368C12-L368C12)

**Comment by GuoxiaWang** [comment]
问题已复现，并且我复现出来的是gpu上的，跟cpu还不一样，我们进一步跟进修复。感谢反馈。
